modules/codegen-plugin/src/smithy4s/codegen/Smithy4sCodegenPlugin.scala

```diff
@@ -134,9 +199,78 @@ object Smithy4sCodegenPlugin extends AutoPlugin {
       artifactFile <- module.artifacts
     } yield {
       val (_, file) = artifactFile
-      os.Path(file)
+      file
+    }
+
+  private def moduleIdEncode(
+      moduleId: ModuleID,
+      scalaBinaryVersion: Option[String]
+  ): List[String] = {
+    (moduleId.crossVersion, scalaBinaryVersion) match {
+      case (Disabled, _) =>
+        List(s"${moduleId.organization}:${moduleId.name}:${moduleId.revision}")
+      case (_: Binary, Some(sbv)) =>
+        List(
+          s"${moduleId.organization}:${moduleId.name}_${sbv}:${moduleId.revision}"
+        )
+      case (_, _) => Nil
+    }
+  }
+
+  /**
+   * Retrieves the smithy4sDependencies that compile-dependencies may have listed
+   * in their jar manifests when they were packaged.
+   */
+  private def smithy4sFetchUpstreamCodegenDependencies(
+      config: Configuration
+  ): Def.Initialize[Task[Seq[File]]] =
+    Def.task {
+      val smithy4sDependencies =
+        (config / externalDependencyClasspath).value
+          .map(_.data)
+          .flatMap(extract)
+      def getJars(ids: Seq[ModuleID]): Seq[File] = {
+        val syntheticModule =
+          organization.value % (name.value + "-smithy4s-resolution") % version.value
+        val depRes = (update / dependencyResolution).value
+        val updc = (update / updateConfiguration).value
+        val uwconfig = (update / unresolvedWarningConfiguration).value
+        val modDescr = depRes.moduleDescriptor(
+          syntheticModule,
+          ids.toVector,
+          None
+        )
+
+        depRes
+          .update(
+            modDescr,
+            updc,
+            uwconfig,
+            streams.value.log
+          )
+          .map(_.allFiles)
+          .fold(uw => throw uw.resolveException, identity)
+      }
+      getJars(smithy4sDependencies)
     }
 
+  private lazy val simple = raw"([^:]*):([^:]*):([^:]*)".r
+  private lazy val cross = raw"([^:]*)::([^:]*):([^:]*)".r
```

Comment:
just spitballing, could we use coursier's api to parse the dependency strings?

Reply:
dunno, if it was a structured parsing result we could probably do that - if there's no way then the answer to my question is "no" ;)

Reply:
Let's postpone this : right now the problem is more about "can we you render a dependency as a coursier-parsable string in both SBT and mill ?" Because I couldn't find a way to do that easily, I have the control over the format and parsing with coursier is not necessary as we're falling in the simplest possible cases (normal java dep, normal scala dep). 

If it proves to be limitative we'll revise.